ckanext/canada/tests/test_validators.py

```diff
@@ -401,17 +401,70 @@ def test_validation_schema(self):
             'schema': 'https://www.annakarenina.com'
         }
 
-        with pytest.raises(ValidationError) as ve:
-            self.sysadmin_action.resource_create(**resource_data)
-        err = ve.value.error_dict
-        assert 'schema' in err
-        assert 'Schema URLs are not supported' in err['schema'][0]
+        resource = self.sysadmin_action.resource_create(**resource_data)
+        assert resource['schema'] == None
 
         resource_data = dict(resource_data,
                             schema='{"fields":["this is bad JSON for Schema"]}')
 
-        with pytest.raises(ValidationError) as ve:
-            self.sysadmin_action.resource_create(**resource_data)
-        err = ve.value.error_dict
-        assert 'schema' in err
-        assert 'Invalid JSON for Schema' in err['schema'][0]
+        resource = self.sysadmin_action.resource_create(**resource_data)
+        assert resource['schema'] == None
+
+
+    def test_validation_options(self):
+        "creating a resource with lax validation options should remove them"
+        pkg = self.sysadmin_action.package_create(**self.complete_pkg)
+
+        resource_data = {
+            'name_translated': {'en': u'Full text.', 'fr': u'Full text.'},
+            'format': u'TXT',
+            'url': u'http://www.annakarenina.com/download/',
+            'size': 42,
+            'resource_type': 'dataset',
+            'language': ['zxx'],
+            'package_id': pkg['id'],
+            'validation_options': {
+                'skip_checks': [],
+                'headers': 3,
+                'scheme': 'http',
+                'format': 'PNG',
+                'encoding': 'ascii',
+                'compression': True,
+                'pick_rows': [5, 6, 8],
+                'skip_rows': [3, 4],
+                'pick_fields': ['A', 'B'],
+                'skip_fields': ['C'],
+                'allow_html': True,
+                'post_parse': 'lambda',
+                'custom_loaders': 'lambda',
+                'custom_parsers': 'lambda',
+                'delimiter': ';',
+                'sheet': 2,
+                'fill_merged_cells': True,
+                'workbook_cache': 'lambda',
+                'row_limit': 'STR to 1 MIL'
+            }
+        }
+
+        resource = self.sysadmin_action.resource_create(**resource_data)
+        assert resource['validation_options'] == {'row_limit': 1000000,
+                                                  'dialect': {
```

Comment:
I don't think it makes sense to export this into the metadata of every resource for every dataset on our site considering it's a fixed value. Use a configuration option instead?

I don't see the utf-8 encoding here, did we force that in another part of the code?

Reply:
@wardi Okay sounds good, I suppose we can remove the stuff from here really. I will just always return blank values for these fields.

And then in `ckanext-validation` we have our our fork/branch so will just hardcode 1M rows in there with a config option?

And then as Validation and Xloader both use the dialects, I will make a config option like `ckanext.canada.ds_resource_dialects` and have that be used in Xloader and Validation for static dialects.

As for the encoding, we do not have static encoding, goodtables/xloader/validation will try to guess encoding and decode it. So I suppose I will do another config option like `ckanext.canada.ds_resource_encoding` and use that value in Xloader and Validation.

This way we will get our goodtables/frictionless to use the `utf-8` encoding and static, basic dialects for TSV and CSV.

Reply:
That sounds good, and instead of `canada` this and that it could be a proper ckanext-validation feature that lets you set static dialects and encoding instead of detecting them. I imagine there are others that would be interested in standardizing their csv files.

Reply:
@wardi okay so we now have these PRs:
- https://github.com/open-data/ckanext-validation/pull/11
- https://github.com/open-data/ckanext-xloader/pull/20

Which will use the new `ckanext.validation.static_validation_options` config. That will handle the static encoding and static dialects that goodtables and tabulator expects.

And then we just parse the same stuff in the Xloader fork to use the same static encoding and dialects.